app/models/review.rb

```diff
@@ -0,0 +1,9 @@
+class Review < ApplicationRecord
+  validates :rating, presence: true
+  validates :body, presence: true
+  validates :user_id, presence: true
+  validates :shop_id, presence: true
```

Comment:
in Rails, a `belongs_to` association is required by default, meaning we don't need to validate the foreign key ids!